api/src/Controllers/AssignController.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace SynchWeb\Controllers;
+
+use Slim\Slim;
+
+use SynchWeb\Page;
+use SynchWeb\Model\Services\AssignData;
+
+class AssignController extends Page
+{
+    private $assignData;
+
+    public static $arg_list = array('visit' => '\w+\d+-\d+', 'cid' => '\d+', 'did' => '\d+', 'pos' => '\d+', 'bl' => '[\w\-]+');
+
+    public static $dispatch = array(array('/visits(/:visit)', 'get', 'getBeamlineVisits'),
+            array('/assign', 'get', 'assignContainer'),
+            array('/unassign', 'get', 'unassignContainer'),
+            array('/deact', 'get', 'deactivateDewar'),
+            array('/names', 'get', 'getPuckNames'),
+
+    );
+
+    function __construct(Slim $app, $db, $user, AssignData $assignData)
+    {
+        // Call parent constructor to register our routes
+        parent::__construct($app, $db, $user);
+        $this->app = $app;
+        $this->assignData = $assignData;
+    }
+
+    function assignContainer()
+    {
+        $cs = $this->assignData->getContainer($this->arg('visit'), $this->arg('cid'));
+
+        if (sizeof($cs) > 0)
+        {
+            $this->assignData->assignContainer($cs[0], $this->arg('pos'));
+            $this->_output(1);
+        }
+        else
+        {
+            $this->_output(0);
+        }
+    }
+
+    function unassignContainer()
```

Comment:
Test history displays if unassign container because it now has a blank string in

Reply:
Tested,, display is good